tests/phpunit/PaymentMethods/WC_Stripe_Express_Checkout_Helper_Test.php

```diff
@@ -984,4 +985,46 @@ public function provide_test_has_free_trial() {
 			],
 		];
 	}
+
+	/**
+	 * Test that OPC detection logic works correctly.
+	 *
+	 * @dataProvider provide_opc_detection_scenarios
+	 *
+	 * @param bool  $is_opc Whether OPC function returns true.
+	 * @param array $button_locations Button location settings.
+	 * @param bool  $should_allow_checkout Expected result for checkout location check.
+	 *
+	 * @return void
+	 */
+	public function test_opc_detection_logic( $is_opc, $button_locations, $should_allow_checkout ) {
+		$stripe_settings                                           = WC_Stripe_Helper::get_stripe_settings();
+		$stripe_settings['payment_request_button_locations']       = $button_locations;
+		WC_Stripe_Helper::update_main_stripe_settings( $stripe_settings );
+
+		$gateway = $this->getMockBuilder( WC_Stripe_UPE_Payment_Gateway::class )
+			->disableOriginalConstructor()
+			->getMock();
+
+		$helper = new WC_Stripe_Express_Checkout_Helper( $gateway );
+
+		// The actual test: Check if should_show_ece_on_checkout_page returns the expected value.
+		$result = $helper->should_show_ece_on_checkout_page();
+
+		$this->assertEquals( in_array( 'checkout', $button_locations, true ), $result );
```

Comment:
This feels like a really complex set of logic to me. Can we split out the logic so it's a bit more explicit as to what we are testing?
```suggestion
		$should_show_ece = in_array( 'checkout', $button_locations, true );
		$this->assertEquals( $should_show_ece, $result );
```

Reply:
This test was updated in follow-up commits